frontend/src/metabase/components/Header.jsx

```diff
@@ -1,28 +1,71 @@
+/* @flow */
 import React, { Component } from "react";
-import ReactDOM from "react-dom";
+import { findDOMNode } from "react-dom";
 
-import Input from "metabase/components/Input.jsx";
-import HeaderModal from "metabase/components/HeaderModal.jsx";
-import TitleAndDescription from "metabase/components/TitleAndDescription.jsx";
-import EditBar from "metabase/components/EditBar.jsx";
+import Input from "metabase/components/Input";
+import HeaderModal from "metabase/components/HeaderModal";
+import TitleAndDescription from "metabase/components/TitleAndDescription";
+import EditBar from "metabase/components/EditBar";
 
 import { getScrollY } from "metabase/lib/dom";
 
+type Props = {
+    // Buttons to be displayed in the header in default mode
+    defaultActions: [],
+
+    // Actions to be displayed in the header in edit mode
+    editingActions: [],
+
+    // Actions to be displayed in the header if in fullscreen mode
+    fullscreenActions: [],
+
+    // Whether or not editing is taking place
+    isEditing: bool,
+
+    // Whether or not fullscreen is active
+    isFullscreen: bool,
+
+    // What is this for and why?
```

Comment:
pls explain :)

Reply:
Haha I added this cause I was confused myself. It seems like it's meant to distinguish between cases where the header is somehow in edit mode but maybe at one point there was a distinction between that and actually showing the text inputs to change the title and description. Right now both `isEditing` and `isEditingInfo` seem to get their value from the same variable in `DashboardHeader` so this was more of a note of "Do we even need this distinction?" I'd maintain that we don't.